server/form-pages/apply/risk-management-features/convictedOffences.ts

```diff
@@ -31,7 +31,7 @@ export default class ConvictedOffences implements TasklistPage {
   }
 
   next() {
-    return 'date-of-offence'
+    return this.body.response === 'yes' ? 'type-of-convicted-offence' : ''
```

Comment:
I feel like I can reason that this is fine, but I can't picture what happens if it there's more non-optional pages after this, where 'previous' would need to be dynamic (me thinking out-loud, not anything you need to action in this PR)

Reply:
That's a good point and that will be happening as part of future work so thanks for raising, I'll have a think